### PR TITLE
Create bazel builds

### PR DESCRIPTION
**Pull request by dominichamon**
As per #191 adding some basic bazel BUILD files. This will require some extra maintenance if files are removed/added but it might become the default some day. For now it's useful to have for interacting with other projects that already use bazel.

### CODE REVIEW

File: test/BUILD [hunk 1]
```diff
@@ -0,0 +1,105 @@
+cc_library(
+    name = "output_test_helper",
+    srcs = [
+        "output_test.h",
+        "output_test_helper.cc",
+    ],
+    deps = ["//:benchmark"],
+)
+
+cc_binary(
```
Comment (nlkm) [inline test/BUILD]: Is it possible to use cc_test instead?

Shouldn't users be able to do "bazel test :all" to validate that all tests are passing?

Comment (dominichamon) [inline test/BUILD]: I wish they could, but we don't run googletest tests (which cc_test expects, as i understand it).

File: BUILD [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    hdrs = [
+        "include/benchmark/benchmark.h",
+        "include/benchmark/benchmark_api.h",
+        "include/benchmark/macros.h",
+        "include/benchmark/reporter.h",
+    ],
+    visibility = ["//test:__pkg__"],
+    defines = ["HAVE_POSIX_REGEX"],
+    includes = ["include"],
+    linkopts = ["-pthread"],
```
Comment (EricWF) [inline BUILD]: Shouldn't we be adding -std=c++11 somewhere?

File: test/BUILD [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    srcs = ["complexity_test.cc"],
+    deps = [
+        ":output_test_helper",
+        "//:benchmark",
+    ],
+    linkopts = ["-lm", "-pthread"],
+)
+
+# cc_binary(
+#     name = "cxx03_test",
```
Comment (EricWF) [inline test/BUILD]: This test needs an explicit `-std=c++03`.

**Comment by EricWF** [comment]
@dominichamon What's the plan with this?

**Comment by dominichamon** [comment]
on its one year anniversary i plan on reverting it.

actually i should probably just do it now. noone is screaming out for it, and this closed PR will be in the history for anyone that wants to do it.

**Comment by shakeel** [comment]
Abseil.io mandates using Bazel for building code, it would be so nice if benchmark would have Bazel BUILD files, it would make the task of using Benchmark http_archive so much easier.

**Comment by dominichamon** [comment]
i speak to the abseil folks quite regularly, and they're considering alternatives to bazel to make integration with existing projects easier. Ie, this issue isn't as high priority as i originally expected.

**Comment by g-easy** [comment]
cctz inlines a BUILD file using new_http_archive: https://github.com/google/cctz/blob/master/WORKSPACE

**Comment by keir** [comment]
Just wanted to add to this bug: Having a supported BUILD would be great, so that you can transitively include Google Benchmark easily in other projects. It seems strictly worse to have a multitude of projects copying and pasting their own forked version of a Google Benchmark BUILD file.

I am working to integrate Google Benchmark with Ceres Solver, and will end up copying and pasting a BUILD into the WORKSPACE file. This feels unsatisfying, especially since Bazel makes it natural and easy to automatically download and build dependencies.

Would a patch to add a BUILD file back be accepted?

**Comment by keir** [comment]
Now landed: https://github.com/ceres-solver/ceres-solver/commit/22fa21c6f824ebefae12a9a2fa459714907acc57

**Comment by dominichamon** [comment]
There are a few patches in progress adding BUILD files, but none of them
are complete. Feel free to work with the authors to help them finish the
job :)

Dominic Hamon | Google
*There are no bad ideas; only good ideas that go horribly wrong.*

**Pull request closed without merging**